### PR TITLE
feat: product templates

### PR DESCRIPTION
**Pull request by robot9706**
Agent:
- Traefik conflicted with custom network names - both works now
- Added Traefik ports to config

Crux:
- Added templates which can be defined using JSON
- Added Strapi (with Postgres) template

### CODE REVIEW

File: web/crux/templates/strapi.json
Comment (polaroi8d) [inline web/crux/templates/strapi.json:91]: Can't we use enums?

File: web/crux/src/services/template.file.service.ts
Comment (polaroi8d) [inline web/crux/src/services/template.file.service.ts:39]: any?

File: web/crux-ui/src/validations/template.ts
Comment (polaroi8d) [inline web/crux-ui/src/validations/template.ts:6]: Add product type and description here too, copy description from template desc.

File: web/crux/proto/crux.proto
Comment (robot9706) [reply web/crux/proto/crux.proto:965]: Currently the template contains the type of product , but I can refactor so that the template contains only images then it'd possible to define what type of product to create.

Comment (polaroi8d) [reply web/crux/proto/crux.proto:965]: Thanks.

File: web/crux/templates/strapi.json
Comment (polaroi8d) [reply web/crux/templates/strapi.json:148]: Got it.

File: web/crux/templates/strapi.json
Comment (robot9706) [reply web/crux/templates/strapi.json]: Fixed 😁

**Comment by polaroi8d** [comment]
Found some issues and talked in person. Waiting for the approval from @robot9706 side.

**Review by polaroi8d — APPROVED** [review]
LGTM

File: web/crux-ui/src/models/template.ts
Comment (m8vago) [inline web/crux-ui/src/models/template.ts:9]: you can extend this type from the upper one

File: web/crux/src/app/product/product.service.ts
Comment (m8vago) [inline web/crux/src/app/product/product.service.ts:142]: move this to const instead

File: web/crux/src/services/template.file.service.ts
Comment (m8vago) [inline web/crux/src/services/template.file.service.ts:56]: promisify()

**Review by m8vago — APPROVED** [review]

**Pull request merged**
